easybuild/framework/easyblock.py

```diff
@@ -698,11 +702,18 @@ def obtain_file(self, filename, extension=False, urls=None, download_filename=No
         :param download_filename: filename with which the file should be downloaded, and then renamed to <filename>
         :param force_download: always try to download file, even if it's already available in source path
         :param git_config: dictionary to define how to download a git repository
+        :param download_instructions: instructions to manually add source (used for complex cases)
+        :param alt_location: alternative location to use instead of self.name
         """
         srcpaths = source_paths()
 
         update_progress_bar(PROGRESS_BAR_DOWNLOAD_ALL, label=filename)
 
+        if alt_location is None:
+            location = self.name
+        else:
+            location = alt_location
+
```

Comment:
Does this meant that it ONLY looks in one of the locations?
And that you need to specify this explicitly for each patch/source?

I was more thinking of having a (possibly list of) global alt_location which it would also use to search for sources/patches.

Reply:
Yes, that's what's there right now. In another scenario with multiple options you would also have to make a choice about where the sources should be downloaded to (and you increase the chance of a name clash for source files).

Reply:
With this approach you will always find the OpenMPI sources in the same place. This will also work for components where you might have lots of different locations for source files.

Reply:
By global I meant easyconfig file global

Reply:
btw, I see no trace of this in the MPITrampoline PR (I may be half blind though)

Reply:
I see what you mean by easyconfig global but for me that still has the same problem about where to store files: what happens in the case of bundles? For me, the least surprising thing is to be explicit. At least we can begin with that and then if someone wants to add additional possibilities they can.

Reply:
yeah, I agree, this is better. I wasn't thinking about the download phase...